core/GlobalState.cc

```diff
@@ -1806,6 +1806,27 @@ unique_ptr<GlobalState> GlobalState::deepCopy(bool keepId) const {
     return result;
 }
 
+unique_ptr<GlobalState> GlobalState::copyForIndex() const {
+    auto result = make_unique<GlobalState>(this->errorQueue, this->epochManager);
+
+    result->initEmpty();
+
+    result->files = this->files;
+    result->fileRefByPath = this->fileRefByPath;
+
+    result->silenceErrors = this->silenceErrors;
+    result->autocorrect = this->autocorrect;
+    result->ensureCleanStrings = this->ensureCleanStrings;
+    result->runningUnderAutogen = this->runningUnderAutogen;
+    result->censorForSnapshotTests = this->censorForSnapshotTests;
+    result->sleepInSlowPath = this->sleepInSlowPath;
+    result->requiresAncestorEnabled = this->requiresAncestorEnabled;
+    result->kvstoreUuid = this->kvstoreUuid;
+    result->pathPrefix = this->pathPrefix;
+
+    return result;
+}
```

Comment:
ooc do you think it's worth factoring this into a helper that's used by `deepCopy`? I almost did something similar for the `--minimize-rbi` change but ended up taking a different approach (no wrong answer)

Reply:
I think this should have comments indicating when to add to this list.

Reply:
I wondered about that. It felt like there was a lot of overlap, but starting from initEmpty() made me hesitate. Maybe we could factor out the common functionality into a local helper that they both use?

For the things that are copied over, I'm going to do another pass through what's in `deepCopy` to make sure that I didn't miss anything. I can add a comment indicating when it's good to add new entries.